### PR TITLE
Changes to allow for commit search

### PR DESCRIPTION
**Pull request by dmadisetti**
See #5 

From a given head, one can now traverse the git commit graph with `commit.children` and `commit.parents`

I threw in a couple other tweaks (v minor), but I can remove them

### CODE REVIEW

**Review by juanbenitopr — APPROVED** [review]
Sorry for the delayed review! and thanks for contributing the project ;)

File: python_git_wrapper/repository.py [hunk 1]
```diff
@@ -9,6 +10,15 @@
 from python_git_wrapper.git_service import GitService
 from python_git_wrapper.status import Status
 
+DELIMITER = '¡|&'
+
+
+def join_flags(flags: List[Tuple[bool, str]]):
+    return "".join([flag for (flagged, flag) in flags if flagged])
```
[the remark this answers is not in the document]
Comment (dmadisetti) [reply python_git_wrapper/repository.py]: Done!

File: python_git_wrapper/repository.py [hunk 2]
```diff
@@ -154,16 +158,41 @@ def change_last_commit_message(self, message: str) -> Commit:
         self.execute(f'commit --amend -m', message)
         return self.last_commit
 
-    def get_commit_by_position(self, position: int) -> Commit:
-        delimeter = '¡|&'
+    def get_commit(self, hash: str) -> Commit:
```
[the remark this answers is not in the document]
Comment (dmadisetti) [reply python_git_wrapper/repository.py:153]: Yeah jut a little bit more general :tada:

**Comment by dmadisetti** [comment]
Threw in CI for fun as well

You might want to update the pypi package as well. I've been using my modded package for a bit with no issues!

**Review by juanbenitopr — APPROVED** [review]

**Comment by dmadisetti** [comment]
Thanks @juanbenitopr ! Are you going to merge this is too?

**Comment by juanbenitopr** [comment]
@dmadisetti Yes! absolutely! and sorry for the delayed approbal, I was mean to approve this a long time ago, thanks a lot for contribute!

**Comment by juanbenitopr** [comment]
@dmadisetti sorry, but I have been busy with the work, I hope to get time to generate the appropriate release, and the CI/CD pipeline in a few months, if you need this for some work, tell me and I will try to prioritize the release work :)

Thanks again for contribute!

**Pull request merged**